webapp/security/views.py

```diff
@@ -212,6 +201,13 @@ def api_create_notice():
             400,
         )
 
+    notice = db_session.query(Notice).get(data["notice_id"])
+    if notice:
+        return (
```

Comment:
So @nottrobin was suggesting doing this with a try exception statement. The only way I have found to do this is with `exc.IntegrityError` but this includes a bunch of other SQL errors, not sure if you know a better way.

Reply:
Yes I was trying to avoid making an unnecessary database call, especially when people are in fact doing everything right. At the very least, we could move this query down into the `except` block, so if we get an error then we double-check whether the record already exists. That way at least this API call would be avoided when there's no error.

Reply:
But to get an exception you have to make a db call, right?

Reply:
If you make a db call and it fails, SQL alchemy will return an exception, and that is what you should catch?

Reply:
Yes, but if that's the case I'm not avoiding any DB calls.

Reply:
Presumably the notices table knows that the `notice_id` is a unique key? This means that when you do `db_session.add(); db_session.commit()` you'll get an error, right? This is where you should catch an exception.

Reply:
If I do it that late then I'll end up doing more db calls, than right now, since I have to query for the release and reference ID's to create the m2m relationships. So by not perfomring that first check I would end up hitting the DB a bunch of times to then have the insert fail on the unique constraint.

Reply:
Okay, but that doesn't seem like the flow we want to optimise for. In most cases, we expect users to understand the data model, and so when they've got as far as submitting a payload in the expected format, you'd think they would also understand that they should be creating a unique USN with a unique USN ID.

The path we want to be performant is the actual successful path where a USN is created. This is because this is the path that it most likely to be scripted and repeated in bulk. In this case this also simplifies the code.

Reply:
I'll udpate